zapisy/apps/enrollment/timetable/assets/components/filters/MultiSelectFilter.vue

```diff
@@ -0,0 +1,316 @@
+      if (isEmpty(selectedIds)) {
+        url.searchParams.delete(this.property);
+      } else {
+        url.searchParams.set(this.property, selectedIds.join(","));
+      }
+      window.history.replaceState(null, "", url.toString());
+
+      this.registerFilter({
+        k: this.filterKey,
+        f: new ExactFilter(selectedIds, this.property),
+      });
+    },
+  },
+});
+</script>
+
+<template>
+  <div class="mb-2">
+    <multiselect
+      v-model="selected"
+      :options="options"
+      :show-labels="showLabels"
+      :multiple="true"
+      :close-on-select="false"
+      :track-by="trackBy"
+      :label="propAsLabel"
+      :placeholder="selectionDescription"
+    >
+      <template slot="option" slot-scope="props">
+        <div class="option-row">
+          <div class="custom-control custom-checkbox">
+            <input
+              type="checkbox"
+              class="custom-control-input"
+              :checked="selected.includes(props.option)"
+            />
+            <label class="custom-control-label" :for="filterKey"></label>
+          </div>
+          {{ props.option.label }}
+        </div>
+      </template>
+      <template slot="selection" slot-scope="{ values, isOpen }">
+        <span class="multiselect__single" v-if="values.length" v-show="!isOpen">
+          {{ selectionDescription }}
+        </span>
+      </template>
+      <template slot="clear">
+        <div
+          v-show="selected.length"
+          class="multiselect__clear"
+          @mousedown.prevent.stop="clearAll"
+        >
+          ×
+        </div>
+      </template>
+    </multiselect>
+  </div>
+</template>
+
+<style src="vue-multiselect/dist/vue-multiselect.min.css"></style>
+
+<style>
+.multiselect__clear {
+  position: absolute;
+  right: 34px;
+  height: 100%;
+  width: 40px;
+  cursor: pointer;
+  z-index: 2;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+}
+
+.multiselect,
+.multiselect__input,
+.multiselect__single {
+  font-size: 0.9rem;
+  white-space: nowrap;
+  overflow: hidden;
+  text-overflow: ellipsis;
+}
+
+.multiselect {
+  display: table;
+  table-layout: fixed;
+  min-height: unset;
+}
+
+.multiselect--active .multiselect__select {
+  transform: none;
+}
+
+.multiselect--active .multiselect__select:before {
+  transform: rotateZ(180deg);
+}
+
+.multiselect__input {
+  padding: 0;
+}
+
+.multiselect__input,
+.multiselect__single {
+  padding: initial;
+  margin-bottom: initial;
+}
+
+.multiselect__tags {
+  min-height: calc(1.5em + 0.75rem + 2px);
+  padding: 0.375rem 4rem 0.375rem 0.75rem;
+}
+
+.multiselect__select {
+  line-height: unset;
+  width: auto;
+  height: auto;
+  right: 0px;
+  top: 0px;
+  padding: 0.375rem 0.75rem;
+  background: rgb(248, 249, 250);
+  border-top-left-radius: 0;
+  border-bottom-left-radius: 0;
+  border-top-right-radius: 5px;
+  border-bottom-right-radius: 5px;
+  border: 1px solid #6c757d;
+}
+
+.multiselect__select::before {
```

Comment:
Rozumiem, że selektor pseudoelementu jest tu bardziej na miejscu niż pseudoklasy, ale mam pewien niepokój np. o kompatybilność, zwłaszcza że importujemy styl, w którym jest to zrobione inaczej (nawet jeśli błędnie). Jak mogę przestać się bać i pokochać bombę?
```suggestion
.multiselect__select:before {
```
Ditto w https://github.com/iiuni/projektzapisy/blob/f4477abbce76ca1bc606bc655f5c9b65e2859ecc/zapisy/apps/enrollment/timetable/assets/components/filters/MultiSelectFilter.vue#L301

Reply:
Mnie również zdziwiła ta tolerancyjna składnia, ale ponieważ nie znałem rozróżnienia między pseudoklasami a pseudoelementami, nie wgłębiałem się.
Do pokochania tego rodzaju bomb najlepsza jest dr. [dokumentacja](https://www.w3.org/TR/selectors/#pseudo-element-syntax): z niej można się dowiedzieć, że styl, z którego importujemy, jest zgodny z CSS2, podczas gdy składnia podwójnego dwukropka pojawiła się w CSS3.0. Element `::before`, jako jeden z obecnych w starszej wersji, otrzymał w nowszej prawo do dwóch składni. Dokumentacja pisze jednak wprost, aby używać nowej składni – stąd wydaje mi się, że powinniśmy zostawić ten kawałek kodu takim, jaki jest obecnie.